js/data/odata/context.d.ts

```diff
@@ -17,7 +31,7 @@ export interface ODataContextOptions {
      * @prevFileNamespace DevExpress.data
      * @public
      */
-    beforeSend?: ((options: { url?: string, async?: boolean, method?: string, timeout?: number, params?: any, payload?: any, headers?: any }) => void);
+    beforeSend?: ((options: { url: string, async: boolean, method: string, timeout: number, params: any, payload: any, headers: any }) => void);
```

Comment:
Not sure all these options are required

Reply:
All the fields of the `options` argument are set before calling the callback, so I decided to make them required